Updated throttle behaviour during takeoff

Georacer opened the pull request:
To include the updates of https://github.com/ArduPilot/ardupilot/pull/27174

Georacer commented on plane/source/docs/automatic-takeoff.rst:
```diff
@@ -12,12 +12,10 @@ ArduPlane 4.0 and later versions.
 ==================
 
-The basic idea of automatic takeoff is for the autopilot to set the
-throttle to maximum and climb until a designated altitude is reached. To
```
I thought this would be explained later in the dedicated section.

Georacer commented on plane/source/docs/automatic-takeoff.rst:
```diff
@@ -26,8 +24,8 @@ for a wide range of aircraft.
-degrees until 5 meters of altitude is reached, gradually allowing more roll until
-the normal :ref:`ROLL_LIMIT_DEG<ROLL_LIMIT_DEG>` limit is allowed once 15 meters of altitude is
+degrees until :ref:`TKOFF_LVL_ALT<TKOFF_LVL_ALT>` (10m by default) meters of altitude is reached, gradually allowing more roll until
+the normal :ref:`ROLL_LIMIT_DEG<ROLL_LIMIT_DEG>` limit is allowed once 3x :ref:`TKOFF_LVL_ALT<TKOFF_LVL_ALT>` meters of altitude is
```
We increased that limit from 5 to 10 as per your request recently @Hwurzburg . Seeing how it causes limited roll up until 3 times that, do you think we should consider reverting to 5m?

Georacer commented on plane/source/docs/automatic-takeoff.rst:
```diff
@@ -238,8 +241,4 @@ general.
-==========================================
-Taking off without an airspeed sensor using a catapult may cause less than maximum throttle to be used due to high initial climb rates. For heavy vehicles, this may result in a stall due to the long time constants used in TECS to adjust throttle after the initial launch. The parameter :ref:`TKOFF_THR_MAX_T<TKOFF_THR_MAX_T>` can be used to force maximum throttle for a time, irrespective of climb r ...
\ No newline at end of file
+Since control effectiveness varies with airspeed, ArduPilot automatically scales the control gains in stabilized modes with airspeed to allow stability at low speeds and to avoid oscillations at high airspeeds. However, when an airspeed sensor is not used, an estimated airspeed based on GPS speed, accelerometer inputs, and position changes is used. During takeoffs into strong head wind, this esti ...
```
Since by default the throttle will be set to maximum, I thought that such catapult launches wouldn't be an issue anymore. Hence I deleted the whole section.

Georacer commented on plane/source/docs/takeoff-mode.rst:
```diff
@@ -36,8 +35,27 @@ highly recommended that a compass be enabled and properly configured for
+
+.. image:: ../images/plane_takeoff_throttle_option_0.png
+
+.. image:: ../images/plane_takeoff_throttle_option_1.png
```
I have the original pictures in DrawIO format, so let me know if you need any changes. They're easy to do.
Or even easier, I can send you the file ;)

Georacer commented:
@Hwurzburg I made an effort in updating the documentation of the takeoff throttle.
Feel free to adapt anything to your liking.

Hwurzburg commented:
@Georacer questions....what happens with option =1 but no airspeed sensor? that is not addressed here...also at one point you say max thr is imposed until lvl_alt,but also say it isnt for nav takeoffs higher up...whats correct?

Georacer commented:
@Hwurzburg I have updated according to the now merged https://github.com/ArduPilot/ardupilot/pull/27758.

> what happens with option =1 but no airspeed sensor?
As I write in `takoff-mode.rst`:
> In case more fine-grained throttle control is required, an airspeed sensor must be installed and enabled via the :ref:`ARPSD_USE<ARSPD_USE>` parameter and :ref:`TKOFF_OPTIONS<TKOFF_OPTIONS>` bit 0 must be set to 1.

In practice, if one does not have an airspeed sensor enabled, the logic will revert back to `TKOFF_OPTIONS=0`.

> also at one point you say max thr is imposed until lvl_alt,but also say it isnt for nav takeoffs higher up.

With the new TECS PR, `TKOFF_LVL_ALT` will be used for both TAKEOFF and AUTO modes. I have updated the documentation accordingly.

Hwurzburg commented:
@Georacer I made some tweaks....I havent squashed everything yet so you could easily see the changes I made...

Hwurzburg commented:
@Georacer George can you resolve thee conflicts and squash and then I will merge, or do you want me to do that?

Georacer commented:
@Hwurzburg I rebased and resolved the conflict.

I didn't find anything actually different. Perhaps it was a space at the end of a line or the lack of a newline.
